tests/data/dwarf_data.h

```diff
@@ -0,0 +1,738 @@
+#pragma once
+
+// The data consists of following source file objects:
+//
+//  struct Foo1 {
+//    int  a;
+//    char b;
+//    long c;
+//  };
+//
+//  struct Foo2 {
+//    int         a;
+//    union {
+//      struct Foo1 f;
+//      struct {
+//        char      g;
+//      };
+//    };
+//  };
+//
+//  struct Foo3 {
+//    struct Foo1 *foo1;
+//    const volatile struct Foo2 * restrict foo2;
+//  };
+//
+//  struct Foo3 foo3;
+//
+//  struct Foo3 *func_1(int a, struct Foo1 *foo1, struct Foo2 *foo2)
+//  {
+//    return 0;
+//  }
+//
+//  struct Foo3 *func_2(int a, int *b, struct Foo1 *foo1)
+//  {
+//    return 0;
+//  }
+//
+//  struct Foo3 *func_3(int a, int *b, struct Foo1 *foo1)
+//  {
+//    return 0;
+//  }
+//
+// int main(void)
+// {
+//   func_1(0, 0, 0);
+//   return 0;
+// }
+
+// generated by:
+//  % cat a.c
```

Comment:
would be nice to have cmake do this

Reply:
Good idea, I tried to do so in the last commit.

Would be nice if we could do the same for BTF data in `tests/data/bft_data.h`. I actually have it prepared, the problem is that it requires pahole version >=1.17 which is not available in our CI environment.